test/Microsoft.AspNet.Mvc.Core.Test/Rendering/HtmlHelperSelectTest.cs

```diff
@@ -0,0 +1,533 @@
+                        Environment.NewLine +
+                        "<option value=\"0\">Zero</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"1\">One</option>" + Environment.NewLine +
+                        "<option value=\"2\">Two</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"3\">Three</option>" + Environment.NewLine +
+                        "</optgroup>" + Environment.NewLine +
+                        "</select>"
+                    },
+                    {
+                        OneDisabledGroupAllSelectedSelectList,
+                        "<select id=\"Property1\" multiple=\"multiple\" name=\"Property1\">" +
+                        "<optgroup disabled=\"disabled\" label=\"Disabled Group\">" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"0\">Zero</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"1\">One</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"2\">Two</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"3\">Three</option>" + Environment.NewLine +
+                        "</optgroup>" + Environment.NewLine +
+                        "</select>",
+                        "<select id=\"Property1\" multiple=\"multiple\" name=\"Property1\">" +
+                        "<optgroup disabled=\"disabled\" label=\"Disabled Group\">" + Environment.NewLine +
+                        "<option value=\"0\">Zero</option>" + Environment.NewLine +
+                        "<option value=\"1\">One</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"2\">Two</option>" + Environment.NewLine +
+                        "<option value=\"3\">Three</option>" + Environment.NewLine +
+                        "</optgroup>" + Environment.NewLine +
+                        "</select>",
+                        "<select id=\"Property1\" multiple=\"multiple\" name=\"Property1\">" +
+                        "<optgroup disabled=\"disabled\" label=\"Disabled Group\">" + Environment.NewLine +
+                        "<option value=\"0\">Zero</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"1\">One</option>" + Environment.NewLine +
+                        "<option value=\"2\">Two</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"3\">Three</option>" + Environment.NewLine +
+                        "</optgroup>" + Environment.NewLine +
+                        "</select>"
+                    },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(DropDownListDataSet))]
+        public void DropDownList_WithNullModel_GeneratesExpectedValue(
```

Comment:
Might want to additionally specify that it does not change the SelectListItems as part of the test name.